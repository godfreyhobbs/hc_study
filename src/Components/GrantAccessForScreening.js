import React, {Component} from 'react';
import '../App.css';
import {ToastContainer, toast} from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';

import { css } from 'glamor';

class GrantAccessForScreening extends Component {

  constructor(props) {
    super(props);

    this.state = {
      dataHash: this.props.searchResultJSON.dataHash,
      viewerAddress: '0x25c63834a9d3cd3c221b7cfbf6b1a02bdffcf0d9',
    };

  }

  handleClick() {
    //dummy dummyDataUri
    toast("You transaction is being processed. This may take several minutes.",{ autoClose: 90000,
      position:'top-center',  bodyClassName: css({
        fontSize: '22px',
      }),
    });
    const dummyDataUri = 'Qmf4vJySpRoexvPgmiokbsC5p3UjNVs38Dz9TvdDdmAqQP';
    this.props.permissionsInstance.grantAccess(this.state.dataHash, this.state.viewerAddress, dummyDataUri, {
      from: this.props.accounts[0],
      gas: 400000,
      gasPrice: 40000000000,
    })
       .then(result => {
         toast.dismiss();
         this.props.callback("ScreeningResults");
       })
       .catch(result => {
         console.error(result);
         toast.dismiss();
         toast("ERROR"+result, { autoClose: 90000,
           position:'top-center',  bodyClassName: css({
             fontSize: '20px',
           })});
       });

  }

  render() {
    return (
      <div>
        <ToastContainer />

        <div className='terms-and-conditions'>
          <p>Some information to grant access to State College to use your health data to screen for study eligibility.</p>
          <p>Data will be used by researches for the study</p>
          <p>The researchers will not retain or share your data</p>
        </div>

        <div className='medical-summary'>
          <div className='FHIR-docs'>FHIR Docs</div>
          <div className='medical-overview'>Medical Record Summary FHIR Docs</div>
          <div className='hospital'>Joe's Hospital</div>
          <div className='FHIR-docs'>Metadata:{this.props.searchResultJSON.metadata}</div>
          <div className='medical-overview'>IRIS Score: {this.props.searchResultJSON.irisScore}</div>
          <div className='medical-overview'>Sig Count: {this.props.searchResultJSON.sigCount}</div>
          <div className='medical-overview'>Created At: {this.props.searchResultJSON.createdAt}</div>
          <div className='medical-overview'>Record DataHash: {this.props.searchResultJSON.dataHash}</div>
        </div>

        <p>I understand and agree to the above statements</p>
        <button onClick={(e) => this.handleClick(e)}>
           I Agree
         </button>
      </div>
    );
  }
}

export default GrantAccessForScreening;
