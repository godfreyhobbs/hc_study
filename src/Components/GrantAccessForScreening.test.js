import React from 'react';
import ReactDOM from 'react-dom';
import GrantAccessForScreening from './GrantAccessForScreening';
import renderer from 'react-test-renderer';
const dummyJSON = {"dataHash":"0x7f92099bdee271c19498a5ff4d19d42c02c84c4b092b7ac2a50bbc0947cc2128","owner":"0xf25aac279a27e94bfb0f84929471e093bb0a63fd","metadata":"Angel Marvin","dataUri":"QmU2xovitopRBVDMnps2pzGtFqFkP54JReJhVoqdzLRhpm","irisScore":1,"sigCount":1,"createdAt":"2018-07-18T16:22:39.340Z","updatedAt":"2018-07-18T16:22:39.340Z"};

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<GrantAccessForScreening searchResultJSON={dummyJSON}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});



it('renders correctly', () => {
  const tree = renderer
.create(<GrantAccessForScreening searchResultJSON={dummyJSON}/>)
   .toJSON();
expect(tree).toMatchSnapshot();
});
