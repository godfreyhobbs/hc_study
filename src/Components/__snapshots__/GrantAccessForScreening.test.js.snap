// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="Toastify"
  />
  <div
    className="terms-and-conditions"
  >
    <p>
      Some information to grant access to State College to use your health data to screen for study eligibility.
    </p>
    <p>
      Data will be used by researches for the study
    </p>
    <p>
      The researchers will not retain or share your data
    </p>
  </div>
  <div
    className="medical-summary"
  >
    <div
      className="FHIR-docs"
    >
      FHIR Docs
    </div>
    <div
      className="medical-overview"
    >
      Medical Record Summary FHIR Docs
    </div>
    <div
      className="hospital"
    >
      Joe's Hospital
    </div>
    <div
      className="FHIR-docs"
    >
      Metadata:
      Angel Marvin
    </div>
    <div
      className="medical-overview"
    >
      IRIS Score: 
      1
    </div>
    <div
      className="medical-overview"
    >
      Sig Count: 
      1
    </div>
    <div
      className="medical-overview"
    >
      Created At: 
      2018-07-18T16:22:39.340Z
    </div>
    <div
      className="medical-overview"
    >
      DataHash: 
      0x7f92099bdee271c19498a5ff4d19d42c02c84c4b092b7ac2a50bbc0947cc2128
    </div>
  </div>
  <p>
    I understand and agree to the above statements
  </p>
  <button
    onClick={[Function]}
  >
    I Agree
  </button>
</div>
`;
